srcs/builtin/builtin_exit.c

```diff
@@ -6,20 +6,76 @@
 /*   By: jjaniec <jjaniec@student.42.fr>            +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2018/04/25 17:21:10 by sbrucker          #+#    #+#             */
-/*   Updated: 2018/08/09 17:11:40 by jjaniec          ###   ########.fr       */
+/*   Updated: 2018/09/16 19:24:02 by jjaniec          ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include <twenty_one_sh.h>
 
-inline void	builtin_exit(char **argv, char **envp, t_exec *exe)
-{
-	(void)argv;
-	(void)envp;
+/*
+** Free shell malloced data before exiting
+*/
 
+static void	exit_free_sh_data(t_exec *exe)
+{
 	if (exe->tmp_envp)
 		ft_free_argv(exe->tmp_envp);
 	if (exe->envp)
 		ft_free_argv(exe->envp);
-	exit(0);
+}
+
+/*
+** Check if a string is only composed of numbers
+*/
+
+static int	str_is_num(char *str)
+{
+	if (!(str && *str))
+		return (1);
```

Comment:
The IF is useless. You call this function once and you have already checked that `str` in not `NULL`. Moreover, if `str[0] == '\0'`, It will not go into the while, but into the next if and still return 1